### PR TITLE
fix: use jsonb instead of json for pgvector

### PR DESCRIPTION
**Pull request by devinburnette**
## Description
When trying to integrate pgvector as a vector store with langchain, I got the following error:

```
2023-10-16 18:27:54.326 EDT [87858] ERROR:  operator does not exist: json @> unknown at character 106
2023-10-16 18:27:54.326 EDT [87858] HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
2023-10-16 18:27:54.326 EDT [87858] STATEMENT:
              SELECT *, embedding <=> $1 as "_distance"
              FROM langchain_pg_embedding
              WHERE cmetadata @> $2
              ORDER BY "_distance" ASC
              LIMIT $3;
```

## The problem
From my brief research, it appears that the `@>` operator only works with jsonb. So I'm trying to figure out how this ever worked in the first place, but I'm not a postgres expert.

## Testing / Reproduced

To test this, I ran the following against my database after langchain created it and this seemed to fix my issue:
```sql
ALTER TABLE langchain_pg_embedding ALTER COLUMN cmetadata SET DATA TYPE jsonb;
```

I'm using this python version of langchain for running my document loaders and storing my embeddings in PG, but then I have a chat client that's using the langchain-js lib to query it, so I happened to then go see if pgvector had the same problem over there, and it doesn't.
https://github.com/langchain-ai/langchainjs/blob/aa9b618786757c301e93caeb3253c56b56d05602/langchain/src/vectorstores/pgvector.ts#L244C38-L244C43

Let me know if the fix isn't as simple as this or if there's a better solution that should be considered like type casting or something else?

cc:
@baskaryan, @eyurtsev, @hwchase17

### CODE REVIEW

File: libs/langchain/langchain/vectorstores/pgvector.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ def __post_init__(
         Initialize the store.
         """
         self._conn = self.connect()
-        # self.create_vector_extension()
+        self.create_vector_extension()
```
Comment (devinburnette) [inline libs/langchain/langchain/vectorstores/pgvector.py:135]: also not sure why this was commented out, but seems like it should be part of the setup as well /cc @hwchase17

Comment (devinburnette) [reply libs/langchain/langchain/vectorstores/pgvector.py:135]: without this, i get the following error:

```
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedObject) type "vector" does not exist
LINE 4:  embedding VECTOR,
                   ^

[SQL:
CREATE TABLE langchain_pg_embedding (
	collection_id UUID,
	embedding VECTOR,
	document VARCHAR,
	cmetadata JSON,
	custom_id VARCHAR,
	uuid UUID NOT NULL,
	PRIMARY KEY (uuid),
	FOREIGN KEY(collection_id) REFERENCES langchain_pg_collection (uuid) ON DELETE CASCADE
)

]
```

Comment (Sbargaoui) [reply libs/langchain/langchain/vectorstores/pgvector.py:135]: The creation query doesn't raise an exception if it fails but rather a warning, proposed fix in #11930

**Comment by eyurtsev** [comment]
Closing in favor of: https://github.com/langchain-ai/langchain/pull/18992

**Comment by VintLin** [comment]
@madanlalit Hi! I noticed you have an open PR #34448 for the same Azure client caching issue in LangChain. I've submitted #35087 which aims to provide a more comprehensive, package-wide solution (including LLMs, Embeddings, and standard OpenAI classes). I've also aligned the credential validation behavior with ChatOpenAI across all these classes. Perhaps we can collaborate or you could take a look at the architectural approach in #35087?

**Pull request closed without merging**